src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -790,6 +790,19 @@ protected virtual Cci.IModuleReference TranslateModule(ModuleSymbol module, Diag
             {
                 Debug.Assert(!needDeclaration);
                 namedTypeSymbol = namedTypeSymbol.TupleUnderlyingType;
+
+                // check that underlying type of a ValueTuple is indeed a value type (or error)
+                // this should never happen, in theory,
+                // but if it does happen we should make it a failure.
+                var declaredBase = namedTypeSymbol.BaseTypeNoUseSiteDiagnostics;
+                if (declaredBase.SpecialType != SpecialType.System_ValueType && !declaredBase.IsErrorType())
```

Comment:
Is there a reason not to simply check `namedTypeSymbol.IsValueType`? Note that `TupleTypeSymbol.IsValueType` already defaults to true if the underlying type is an error type.